src/declaration.c

```diff
@@ -1653,6 +1653,7 @@ void VarDeclaration::setFieldOffset(AggregateDeclaration *ad, unsigned *poffset,
             {
                 const char *s = (t->ty == Tsarray) ? "static array of " : "";
                 ad->error("cannot have field %s with %ssame struct type", toChars(), s);
+                return;
```

Comment:
I found your comment you added here to be helpful in understanding what the change is. You found it helpful enough to add here - why not in the code?

I also don't agree that "the added test case will fail" serves as documentation, as the test case is quite remote from the code.